ikarus/solver/nonlinearsolver/solverinfos.hh

```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: 2021-2023 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+namespace Ikarus {
+  struct NonLinearSolverInformation {
+    explicit operator bool() const { return success; }
+    bool success{false};
+    double residualNorm{0.0};
```

Comment:
You can also use here [`std::numeric_limits<double>::<double>::infinity()`](https://en.cppreference.com/w/cpp/types/numeric_limits/infinity) the same for `int`. I think this would be a better default.

Reply:
https://godbolt.org/z/dK3djGWqG - `std::numeric_limits<int>::infinity()` returns zero.
[std::numeric_limits<int>::has_infinity](https://en.cppreference.com/w/cpp/types/numeric_limits/has_infinity) is `false`. So for `int`, may be 0 is a better default?

Reply:
ah ok. didnt know that. maybe init to -1?